src/createWebStoragePersistor-experimental/index.ts

```diff
@@ -16,10 +16,47 @@ export function createWebStoragePersistor({
   key = `REACT_QUERY_OFFLINE_CACHE`,
   throttleTime = 1000,
 }: CreateWebStoragePersistorOptions): Persistor {
+  //try to save data to storage
+  function trySave(persistedClient: PersistedClient) {
+    try {
+      storage.setItem(key, JSON.stringify(persistedClient))
+    } catch {
+      return false
+    }
+    return true
+  }
+
   if (typeof storage !== 'undefined') {
     return {
       persistClient: throttle(persistedClient => {
-        storage.setItem(key, JSON.stringify(persistedClient))
+        if (trySave(persistedClient) !== true) {
+          const mutations = [...persistedClient.clientState.mutations]
+          const queries = [...persistedClient.clientState.queries]
+          const client: PersistedClient = {
+            ...persistedClient,
+            clientState: { mutations, queries },
+          }
+
+          // clean mutations and try to save
+          while (mutations.shift()) {
+            if (trySave(client)) {
+              return // save success
+            }
+          }
+
+          // sort queries by dataUpdatedAt (oldest first)
+          const sortedQueries = [...queries].sort(
+            (a, b) => a.state.dataUpdatedAt - b.state.dataUpdatedAt
+          )
+          // clean old queries and try to save
+          while (sortedQueries.length > 0) {
+            const oldestData = sortedQueries.shift()
+            client.clientState.queries = queries.filter(q => q !== oldestData)
+            if (trySave(client)) {
+              return // save success
+            }
+          }
```

Comment:
that looks good 👍 I can see that we don't have any tests yet, but since this is getting more complex could you maybe add some? I'm also wondering what would happen if the persistFunction now takes longer than the throttleTime 🤔

Reply:
Thank you,

1. I will take a look at the tests in this project, and try to add one for this function.
2. It will be `serial execution`, because, it's a sync function (blocking the timer).

Reply:
A test file was added in the 3rd commit